### ISSUE TITLE
Add MacOS Notarization to release process

### ISSUE THREAD

**Issue by cube2222** [opentofu/opentofu#649]
Right now we don't have notarization in our release process. That basically means, from the usage perspective, that trying to run tofu will fail on MacOS, and the user will have to go to `Settings => Private & Security` and click on the `allow anyway` button that pops up there. It will then work as normal.

It's worth noting that once we make a Homebrew Core package, then this **won't** be an issue there. That is because Homebrew Core packages build the binary from source locally, rather than downloading a remote prebuilt binary. Locally-built binaries don't need to be notarized.

In other words, I don't believe this to be a critical issue, since I expect MacOS users to use a package manager like Homebrew, but we should still fix it eventually and support notarization. We can achieve this with macos workers on github, but I believe there are CLIs that work on Linux, too.

**Comment by loafoe** [comment]
A more pinpointed workaround:

```shell
xattr -d com.apple.quarantine tofu
```

but definitely 👍🏻 for getting notarization in the build process if the intent is to distribute trusted binaries from Github

**Comment by Apollorion** [comment]
This is incredibly important for folks working with opentofu in a regulated environment that doesnt allow running unnotarized software. I was one of those folks a few months back :)

**Comment by skyzyx** [comment]
This is actually relatively straightforward (compared to how it used to be). You _will_ need a macOS CI node, however. And a modern installation of Xcode on that node. And an Apple Developer Account.

Once those things are available, I'm happy to collaborate with the Owner (only an Owner can do this) of the Apple Developer Account to get all the right things in place. I do this at work/home, and notarize other Go binaries I distribute on the web.
